sabnzbd/filesystem.py

```diff
@@ -1369,6 +1384,50 @@ def pathbrowser(path: str, show_hidden: bool = False, show_files: bool = False)
     return file_list
 
 
+if sys.platform == "linux":
+
+    def write_at_offset(fd: int, data: memoryview, offset: int) -> int:
+        return os.pwrite(fd, data, offset)
+
+elif sys.platform == "darwin":
+    pwrite = sabnzbd.MACOSLIBC.pwrite
+    pwrite.argtypes = [ctypes.c_int, ctypes.c_void_p, ctypes.c_size_t, ctypes.c_longlong]
+    pwrite.restype = ctypes.c_ssize_t
+
+    def __pwrite_mac(fd: int, data, offset: int) -> int:
+        if isinstance(data, memoryview):
+            mv = data
+        else:
+            mv = memoryview(data)
+
+        # Make contiguous copy if needed
+        if not mv.contiguous:
+            mv = memoryview(bytearray(mv))
+
+        # Make writable copy if needed
+        if mv.readonly:
+            mv = memoryview(bytearray(mv))
+
+        buf = (ctypes.c_char * len(mv)).from_buffer(mv)
+        n = pwrite(fd, buf, len(mv), offset)
```

Comment:
Doesn't this also work on macOS? 
Since it mentions "Availability: Linux", and [the docs say](https://docs.python.org/3/library/intro.html#availability): "If not separately noted, all functions that claim “Availability: Unix” are supported on macOS, iOS and Android, all of which build on a Unix core."